site/content/in-dev/unreleased/configuring-polaris-for-production.md

```diff
@@ -23,114 +23,179 @@ type: docs
-```
+### Metastore Configuration
 
-Now, to bootstrap Polaris, run:
+A Metastore Manager should be configured with an implementation that durably persists Polaris
+entities. By default, Polaris uses an in-memory metastore.
 
-```bash
-java -jar /path/to/jar/polaris-service-all.jar bootstrap polaris-server.yml
-```
+> [!WARNING]
+> The default in-memory metastore is not suitable for production use, as it will lose all data
+> when the server is restarted; it is also unusable when multiple Polaris replicas are used.
 
-or in a container:
+To use a durable Metastore Manager, you need to switch to the EclipseLink metastore, and provide
+your own `persistence.xml` file. This file contains details of the database used for metastore
+management and the connection settings. For more information, refer to the [metastore
+documentation]({{% ref "metastores" %}}).
 
-```bash
-bin/polaris-service bootstrap config/polaris-server.yml
+Then, configure Polaris to use your metastore by setting the following properties:
+
+```properties
+polaris.persistence.type=eclipse-link
+polaris.persistence.eclipselink.configuration-file=/path/to/persistence.xml
+polaris.persistence.eclipselink.persistence-unit=polaris
 ```
 
-Afterward, Polaris can be launched normally:
+Where:
 
-```bash
-java -jar /path/to/jar/polaris-service-all.jar server polaris-server.yml
+- `polaris.persistence.type` indicates that we are using the EclipseLink metastore.
+- `polaris.persistence.eclipselink.configuration-file` is the path to the `persistence.xml` file.
+- `polaris.persistence.eclipselink.persistence-unit` is the name of the persistence unit to use (in 
+  case the configuration file has many persistence units).
+
+Typically, in Kubernetes, you would define the `persistence.xml` file as a `ConfigMap` and set the
+`polaris.persistence.eclipselink.configuration-file` property to the path of the mounted file in
+the container.
+
+Be sure to secure your metastore backend since it will be storing credentials and catalog metadata.
+
+### Bootstrapping
+
+Before using Polaris, you must **bootstrap** the metastore manager. This is a manual operation that
+must be performed **only once** in order to prepare the metastore manager to integrate with Polaris.
+
+By default, when bootstrapping a new realm, Polaris will create randomised `CLIENT_ID` and
+`CLIENT_SECRET` for the `root` principal and store their hashes in the metastore backend.
+
+Depending on your database, this may not be convenient as the generated credentials are not stored
+in clear text in the database.
+
+In order to provide your own credentials for `root` principal (so you can request tokens via
+`api/catalog/v1/oauth/tokens`), there are two approaches:
+
+1. Use the Polaris Admin Tool to bootstrap the realm and set its `root`
+   principal credentials.
+2. Set the `root` principal credentials when deploying Polaris for the first time.
```

Comment:
Does it make sense to always ask users to use option 1? We don't have to mention option 2 in that case. I think this will simplify usage, esp. for the first time users.

Reply:
I tend to agree, although I remember that the Helm chart currently has a heavy machinery that allows users to go with option 2. But let's remove option 1 from docs and figure out the Helm chart later (big PR coming...).